### DISCUSSION TITLE
Cannot create LMDB store from server REST API (or workbench UI)

### DISCUSSION THREAD

**Discussion by seralf** [eclipse-rdf4j/rdf4j#4115] [Q&A] [answered]
Hi, 

the RDF4J 4.1.0 release claims "LMDB Store is available in the server and workbench". 

While testing the LMDB store from scala/java api seems to work as expected, the workbench UI and the server REST API seems to still have some problems.

1) the workbench seems to still missing the option for creating a new lmdb from the UI.
Looking at the code, the LMDB options seems not to be in the available ones: https://github.com/eclipse/rdf4j/tree/4.1.0/tools/workbench/src/main/webapp/transformations. I tried to create the entry by myself for testing with just the ID and label (it's not clear to me how to enable the LMDBStore type to be discovered), but it always gives me a "null pointer exception", I suppose I'm missing some required parameters.

2)  Apart from the workbench itself, my tests using the REST API of the server to handle the creation of an LMDB-based repository seems to fail, with the error: `MALFORMED DATA: Supplied repository configuration is invalid: could not read repository config from supplied data`, using for example the following configuration:

```bash
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.
@prefix rep: <http://www.openrdf.org/config/repository#>.
@prefix sr: <http://www.openrdf.org/config/repository/sail#>.
@prefix sail: <http://www.openrdf.org/config/sail#>.
@prefix ms: <http://www.openrdf.org/config/sail/memory#>.


[] a rep:Repository ;
   rep:repositoryID "test-lmdb" ;
   rdfs:label "test LMDB store" ;
   rep:repositoryImpl [
      rep:repositoryType "openrdf:SailRepository" ;
      sr:sailImpl [
        sail:sailType "openrdf:LmdbStore" ;
      ]
   ].
```

I suppose I'm doing something wrong here, can anyone help me to create a working minimal configuration?

Thank you in advance for any siggestions,
Alfredo

**Comment by hmottestad** [comment]
@kenwenzel do you have time to help answer this question?

**Reply by kenwenzel** [reply]
Sorry for my late reply. I try to investigate this next week. Unfortunately, I am also not yet familiar with the workbench and its mechanisms for registering additional repository or SAIL types.

**Reply by hmottestad** [reply]
Ok. Maybe I can fix it. I guess I had thought that it was already done when I was writing the release notes for the 4.1.0 release. We can probably fix this in 4.1.1.

**Comment by hmottestad** [comment, marked as answer]
@seralf Could you try with `sail:sailType "rdf4j:LmdbStore" ;` ? That might work in the mean time.

**Reply by seralf** [reply]
Hi @hmottestad thank you for the suggestion!
Now it works! :-) 

I've changed the configuration to the following one:

```bash
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.
@prefix rep: <http://www.openrdf.org/config/repository#>.
@prefix sr: <http://www.openrdf.org/config/repository/sail#>.
@prefix sail: <http://www.openrdf.org/config/sail#>.

[] a rep:Repository ;
   rep:repositoryID "test-lmdb" ;
   rdfs:label "test LMDB store" ;
   rep:repositoryImpl [
      rep:repositoryType "openrdf:SailRepository" ;
      sr:sailImpl [
        sail:sailType "rdf4j:LmdbStore" ;
   ]
 ] .
 ```

Anyway, in order to use specific configurations (let's say for example something like: `lmdb:tripleIndexes "spoc,posc"`, similar to the native store case) I suppose there should be a dedicated prefix / namespace definition? I can imagine having something like: `@prefix lmdb: <http://www.openrdf.org/config/sail/lmdb#>.`, but I didn't find explicit references to it. (Sorry in advance in case I missed them)

**Reply by hmottestad** [reply]
The only supported setting at the moment is the evaluation factory property.
